client-app/src/App.js

```diff
@@ -1,22 +1,66 @@
-import './App.css';
-import {createBrowserRouter, RouterProvider} from "react-router-dom";
-import RegisterPage from "./views/Register"
-import {Container} from "@mui/material";
+import "./App.css";
+import { BrowserRouter, Route, Routes } from "react-router-dom";
+import RegisterPage from "./pages/Register.jsx";
+import { Container } from "@mui/material";
+import Login from "./pages/Login.jsx";
+import { useState } from "react";
+import Dashboard from "./pages/Dashboard.jsx";
+import ProtectedRout from "./components/ProtectedRout.jsx";
+import UnProtectedRout from "./components/UnProtectedRout.jsx";
 
 function App() {
-    const router = createBrowserRouter([
-        {
-            path: "/",
-            element: <RegisterPage/>
-        }
-    ])
-    return (
-        <div className="App">
-            <Container style={{paddingTop: 5}}>
-                <RouterProvider router={router}/>
-            </Container>
-        </div>
-    );
+  const [authorized, setAuthorized] = useState(false);
+  const [userId, setUserId] = useState();
+
+  useEffect(() => {
+    const authorization = async () => {
+      if (authorized) return;
+      const user = await auth.autorized();
+      if (user) {
+        setAuthorized(true);
+        setUserId(user._id);
+      }
+    };
+    authorization();
+  }, [authorized]);
```

Comment:
```suggestion
```
Why here?
WDYT about using React context for that instead?

read more about it [here](https://react.dev/learn/scaling-up-with-reducer-and-context)